politeiad/backendv2/tstorebe/tstore/anchor.go

```diff
@@ -0,0 +1,569 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package tstore
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"time"
+
+	dcrtime "github.com/decred/dcrtime/api/v2"
+	"github.com/decred/dcrtime/merkle"
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/store"
+	"github.com/google/trillian"
+	"github.com/google/trillian/types"
+	"google.golang.org/grpc/codes"
+)
+
+// TODO handle reorgs. An anchor record may become invalid in the case of a
```

Comment:
Wasn't this fixed in dcrtime?

Reply:
I don't think so.  We talked about it the need for it but I don't think anything was ever added to dcrtime to handle reorgs.

This is something that we I can open an issue for and we can fix after tstore is launched.  The way to handle it is to basically just delete the anchor blob from the tstore backend if a >6 block reorg occurs and use the next anchor in the tlog tree.

Reply:
This needs a resolution when we go live with mainnet.

Reply:
We debated this in matrix and the layers above handle the issue I was worried about.